goal_src/jak2/engine/target/mech_suit/carry-h.gc

```diff
@@ -47,13 +49,577 @@
+(defmethod carry-info-method-9 carry-info ((obj carry-info))
+  (let ((s5-0 (-> obj process 0 node-list data (-> obj joint) bone transform)))
+    (vector-rotate*! (-> obj normal) (-> obj local-normal) s5-0)
+    (vector-matrix*! (-> obj point) (-> obj local-point) s5-0)
+    )
+  0
+  (none)
+  )
+
+(defmethod distance-from-destination carry-info ((obj carry-info) (arg0 carry-info))
+  "Returns the distance from the current `point` and the provided [[carry-info]]'s `point`.
+       Returns `-1.0` if it exceeds the maximum allowed"
+  (let* ((f28-0 (vector-y-angle (vector-! (new 'stack-no-clear 'vector) (-> arg0 point) (-> obj point))))
+         (f30-0 (fabs (deg-diff f28-0 (vector-y-angle (-> obj normal)))))
+         (f28-1 (fabs (deg-diff (+ 32768.0 f28-0) (vector-y-angle (-> arg0 normal)))))
+         (f26-0 (vector-vector-distance (-> obj point) (-> arg0 point)))
+         )
+    (cond
+      ((or (< (-> obj max-distance) f26-0)
+           (< (-> arg0 max-distance) f26-0)
+           (< (-> obj max-angle) f30-0)
+           (or (< (-> arg0 max-angle) f28-1) (zero? (logand (-> obj mode) (-> arg0 mode))))
+           )
+       (if (< (-> obj max-distance) f26-0)
+           (format
+             #t
+             "  ~A ~A failed for this distance ~M ~M~%"
+             (-> obj process 0 name)
+             (-> arg0 process 0 name)
+             f26-0
+             (-> obj max-distance)
+             )
+           )
+       (if (< (-> arg0 max-distance) f26-0)
+           (format
+             #t
+             "  ~A ~A failed for other distance ~M ~M~%"
+             (-> obj process 0 name)
+             (-> arg0 process 0 name)
+             f26-0
+             (-> arg0 max-distance)
+             )
+           )
+       (if (< (-> obj max-angle) f30-0)
+           (format
+             #t
+             "  ~A ~A failed for this angle ~R ~R~%"
+             (-> obj process 0 name)
+             (-> arg0 process 0 name)
+             f30-0
+             (-> obj max-angle)
+             )
+           )
+       (if (< (-> arg0 max-angle) f28-1)
+           (format
+             #t
+             "  ~A ~A failed for other angle ~R ~R~%"
+             (-> obj process 0 name)
+             (-> arg0 process 0 name)
+             f28-1
+             (-> arg0 max-angle)
+             )
+           )
+       (if (zero? (logand (-> obj mode) (-> arg0 mode)))
+           (format
+             #t
+             "  ~A ~A failed for mode ~X ~X~%"
+             (-> obj process 0 name)
+             (-> arg0 process 0 name)
+             (-> obj mode)
+             (-> arg0 mode)
+             )
+           )
+       -1.0
+       )
+      (else
+        (+ f26-0 (* 409.6 (- 32768.0 f30-0)))
+        )
+      )
+    )
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod drag! carry-info ((obj carry-info) (arg0 carry-info))
+  (let ((v1-0 (-> arg0 process)))
+    (set! (-> obj other) (the-as handle (logior (if v1-0
```

Comment:
darn, looks like process -> handle isn't working here. But at least this code will work for now.

Reply:
https://github.com/open-goal/jak-project/issues/1883